### PR TITLE
Compare shodan censys

### PR DESCRIPTION
**Pull request by Francisco-aragao**
Fix scripts to filter and analyze ips, ports and scanning frequencies for shodan and censys.

### CODE REVIEW

File: analysisDataset/analysis_shodan_censys_data.py [hunk 1]
```diff
@@ -6,35 +6,44 @@
...
 MODULE_FIELD_IN_SHODAN = "module"
 PREFIX_MODULE_FIELD_IN_SHODAN = "_shodan"
 
+
+"""
+    Class to handle functions to analyze Shodan and Censys data, reading 
+    information, organizing the data and then storing the .json file with the 
+    results. The defined functions are used in the AnalysisShodanCensysData class 
+    in the 'temporal_scan' and 'probe_data' functions.
+"""
```
Comment (cunha) [inline analysisDataset/analysis_shodan_censys_data.py:26]: Please follow the standard for creating docstrings: https://peps.python.org/pep-0257/

File: analysisDataset/analysis_shodan_censys_data.py [hunk 2]
```diff
@@ -6,35 +6,44 @@
...
+
     def __init__(self):
         self.days_scan: dict[str, dict] = {}
         self.unique_ips: list[int] = [0]
         self.all_ips: list[str] = []
         self.repeated_ip_scan: list[int] = [0]
         self.ip_scaned_again_same_day: list[int] = [0]
         self.ips_scanned: list[int] = [0]
-        self.attributes_collected: list[str] = ['']
+        self.attributes_collected: list[str] = [""]
```
Comment (cunha) [inline analysisDataset/analysis_shodan_censys_data.py:38]: Why add an empty element to the list? If an empty list does not work, write a comment mentioning why. (Or try to refactor the code such that an empty list works.)

File: analysisDataset/analysis_shodan_censys_data.py [hunk 3]
```diff
@@ -51,22 +60,25 @@ def add_info_temporal_scan(self, ip: str, timestamp: datetime, index: int):
...
-    
-    def add_info_probe_data(self, ip: str, port:str, scan_attributes: str, module: str, cpe: list | None):
+
+    def add_info_probe_data(
+        self,
+        ip: str,
+        port: str,
+        scan_attributes: str,
+        module: str,
+        cpe_list: list | None,
```
Comment (cunha) [inline analysisDataset/analysis_shodan_censys_data.py:70]: Why do we need `None`? Can't we just use an empty list to capture cases where there is no associated CPE?

File: analysisDataset/analysis_shodan_censys_data.py [hunk 4]
```diff
@@ -51,22 +60,25 @@ def add_info_temporal_scan(self, ip: str, timestamp: datetime, index: int):
...
 
         if not (scan_attributes) in self.attributes_collected:
             self.attributes_collected.append(scan_attributes)
 
-        if (ip) in self.cpe_by_ip:
-            if cpe:
-                for j in cpe:
-                    self.cpe_by_ip[ip].add(j)
-        else:
+        if (ip) not in self.cpe_by_ip:
```
Comment (cunha) [inline analysisDataset/analysis_shodan_censys_data.py:76]: Remove parenthesis around `(ip)`.

File: analysisDataset/analysis_shodan_censys_data.py [hunk 5]
```diff
@@ -104,15 +118,19 @@ def dump_info_temporal_scan(self, output_directory: str, initial_date, final_dat
...
         with open(output_path, "w") as file:
             json.dump(data, file, indent=6)
-    
+
     def dump_info_probe_data(self, output_directory: str):
 
         # store info collected with parentheses
-        self.attributes_collected = [(keysDict) for keysDict in self.attributes_collected]
+        self.attributes_collected = [
+            (keysDict) for keysDict in self.attributes_collected
```
Comment (cunha) [inline analysisDataset/analysis_shodan_censys_data.py:132]: Spurious parentheses.

**Review by cunha — CHANGES_REQUESTED** [review]

File: analysisDataset/analysis_shodan_censys_data.py (hunk 5 above)
Comment (Francisco-aragao) [reply analysisDataset/analysis_shodan_censys_data.py:132]: fixed -> removed parenthesis

File: analysisDataset/analysis_shodan_censys_data.py (hunk 4 above)
Comment (Francisco-aragao) [reply analysisDataset/analysis_shodan_censys_data.py:76]: fixed -> removed parenthesis

File: analysisDataset/analysis_shodan_censys_data.py (hunk 3 above)
Comment (Francisco-aragao) [reply analysisDataset/analysis_shodan_censys_data.py:70]: fixed -> Changed 'None' to empty list

File: analysisDataset/analysis_shodan_censys_data.py (hunk 2 above)
Comment (Francisco-aragao) [reply analysisDataset/analysis_shodan_censys_data.py:38]: fixed -> empty element ('') was irrelevant. Removed

File: analysisDataset/analysis_shodan_censys_data.py (hunk 1 above)
Comment (Francisco-aragao) [reply analysisDataset/analysis_shodan_censys_data.py:26]: fixed -> docstrings

**Pull request closed without merging**